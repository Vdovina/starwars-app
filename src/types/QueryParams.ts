export type QueryParams = {
  page: number | number[];
} & AdditionalQueryParams;

export interface AdditionalQueryParams {
  searchParam?: string;
}
