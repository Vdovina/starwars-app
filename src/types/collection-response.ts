export interface CollectionResponse<T> {
  count: number;
  next: string | null;
  previous: string | null;
  results: T[];
}

export type MultipleCollectionResponse<T> = Pick<CollectionResponse<T>, 'count' | 'results'>;
