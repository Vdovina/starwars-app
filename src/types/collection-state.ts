export type CollectionState<T> = SuccessCollectionState<T> | ErrorCollectionState;

interface SuccessCollectionState<T> {
  data: T[];
  total: number;
}

interface ErrorCollectionState {
  error: any;
}
