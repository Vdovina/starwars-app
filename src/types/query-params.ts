export type SingleListQueryParams = {
  page?: number;
} & AdditionalQueryParams;

export type MultipleListQueryParams = {
  pages: number[];
} & AdditionalQueryParams;

export interface AdditionalQueryParams {
  searchParam?: string;
}
