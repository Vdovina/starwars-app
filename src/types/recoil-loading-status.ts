import { LoadingStatus } from '../constants/statuses';

export type RecoilLoadingStatus = LoadingStatus.Loading | LoadingStatus.Error | LoadingStatus.Success;
