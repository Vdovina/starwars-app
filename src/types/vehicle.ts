export interface Vehicle {
  name: string;
  model: string;
  cargo_capacity: string;
  consumables: string;
  cost_in_credits: string;
  crew: string;
  length: string;
  manufacturer: string;
  max_atmosphering_speed: string;
  vehicle_class: string;
  passengers: string;
  pilots: string[];
  films: string[];
  created: string;
  edited: string;
  url: string;
}
