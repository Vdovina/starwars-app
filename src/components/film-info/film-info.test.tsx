import { create, act as testAct, ReactTestRenderer } from 'react-test-renderer';
import { FilmInfo } from './';
import { Film } from '../../types/film';

const film: Film = {
  title: 'Title',
  episode_id: 1,
  opening_crawl: 'Caption',
  director: 'DirectorName',
  producer: 'ProducerName',
  release_date: '2000-01-01',
  characters: ['some-url/1/'],
  planets: ['some-url/1/'],
  starships: ['some-url/2/'],
  vehicles: ['some-url/4/'],
  species: ['some-url/1/'],
  created: '2000-01-01',
  edited: '2000-01-01',
  url: 'some-url/1/',
};

describe('FilmInfo Tests', () => {
  it('FilmInfo render', () => {
    let component: ReactTestRenderer;
    testAct(() => {
      component = create(<FilmInfo film={film} />);
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
