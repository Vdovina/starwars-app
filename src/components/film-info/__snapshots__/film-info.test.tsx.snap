// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmInfo Tests render 1`] = `
<div
  className="block film-info"
>
  <div
    className="film-info__title"
  >
    Title
  </div>
  <div
    className="film-info__info"
  >
    <div
      className="field__name"
    >
      Episode:
    </div>
    <div
      className="field__value"
    >
      1
    </div>
  </div>
  <div
    className="film-info__info"
  >
    <div
      className="field__name"
    >
      Director:
    </div>
    <div
      className="field__value"
    >
      DirectorName
    </div>
  </div>
  <div
    className="film-info__info"
  >
    <div
      className="field__name"
    >
      Producers:
    </div>
    <div
      className="field__value"
    >
      ProducerName
    </div>
  </div>
  <div
    className="film-info__info"
  >
    <div
      className="field__name"
    >
      Release date:
    </div>
    <div
      className="field__value"
    >
      2000-01-01
    </div>
  </div>
  <div
    className="film-info__summary"
  >
    Caption
  </div>
  <div
    className="film-info__button"
    test-id="Link"
  >
    <svg
      fill="none"
      height="25"
      viewBox="0 0 9 25"
      width="9"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L8 12.5L1 24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
</div>
`;
